src/beta/resources/referral.js

```diff
@@ -0,0 +1,137 @@
+/* eslint-disable no-dupe-class-members,no-unused-vars,class-methods-use-this */
+import superagent from 'superagent';
+
+import T from 'proptypes';
+import base from '../../resources/base';
+import { propTypes as userPropTypes } from '../../resources/user';
+import RequestError from '../../errors/request';
+import API from '../../easypost';
+
+export const propTypes = Object.assign({}, userPropTypes, {
+  name: T.string.isRequired, // mark as required
+  email: T.string.isRequired, // mark as required
+  phone: T.string.isRequired, // handle discrepancy between phone (create Referral) and phone_number (all User functions, retrieve Referral)
+});
+
+export default (api) =>
+  class Referral extends base(api) {
+    static _name = 'Referral';
+
+    static _url = 'referral_customers';
+
+    static key = 'user';
+
+    static propTypes = propTypes;
+
+    static #stripeCreditCardUrl = 'https://api.stripe.com/v1/tokens';
+
+    static #getReferralApi(referralApiKey) {
```

Comment:
```suggestion
    static #getReferralApiKey(referralApiKey) {
```

Reply:
No, this gets an instance of the API with the Referral user's API key populated.

Reply:
`api` was a terrible naming convention, whoever chose that years ago...

Let's document this then as to what it does. I still don't love the name. Could it be `getReferralApiClient` or something to that effect?